loader/windows/icd_windows.c

```diff
@@ -103,19 +171,62 @@ BOOL CALLBACK khrIcdOsVendorsEnumerate(PINIT_ONCE InitOnce, PVOID Parameter, PVO
                 KHR_ICD_TRACE("Value not zero, skipping\n");
                 continue;
             }
-
             // add the library
-            khrIcdVendorAdd(cszLibraryName);
+            status |= adapterAdd(cszLibraryName, ZeroLuid);
```

Comment:
I'm not sure about the semantics here.
Do we want to set a valid status if *any* of the adapter were successfully added or iff all of them succeeded? (i.e. OR vs AND).

Reply:
> 
> 
> Looks better, thanks for the fixes.
> I'm still curious about when do we release the resources (i.e. the dynamically allocated list that gets incremented by 2 in its size), I haven't seen the free for it (i.e. DllMain / callback from the OS on library unload.

The older list gets freed up as soon as new list with twice the size is allocated in adapterAdd.
The final list gets freed up in free after call to adapterFree once all adapters are enumerated.

Reply:
In which file? I don't see the call to free all adapters after adapterFree.

Reply:
> 
> 
> In which file? I don't see the call to free all adapters after adapterFree.

It's is the same file icd_windows.c It's just a single free on pWinAdapterBegin as this is single malloced memory.